File: medplum-java/medplum-model/src/main/java/com/fhir/NutritionOrder.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableNutritionOrder.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface NutritionOrder {
  Optional<dateTime> dateTime();

  Optional<List<CodeableConcept>> foodPreferenceModifier();

  Reference patient();

  Optional<List<uri>> instantiates();

  Optional<List<canonical>> instantiatesCanonical();

  Optional<List<CodeableConcept>> excludeFoodModifier();

  Optional<NutritionOrder_OralDiet> oralDiet();

  Optional<List<Extension>> extension();

  String resourceType();

  Optional<id> id();

  Optional<Reference> orderer();

  Optional<List<NutritionOrder_Supplement>> supplement();

  Optional<List<ResourceList>> contained();

  Optional<List<Identifier>> identifier();

  Optional<List<uri>> instantiatesUri();

  Optional<List<Annotation>> note();

  Optional<NutritionOrder_EnteralFormula> enteralFormula();

  Optional<Reference> encounter();

  Optional<code> language();

  Optional<code> intent();

  Optional<code> status();

  Optional<uri> implicitRules();

  Optional<Meta> meta();

  Optional<Narrative> text();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Reference>> allergyIntolerance();

  static ImmutableNutritionOrder.PatientBuildStage builder() {
    return ImmutableNutritionOrder.builder();
  }
}

File: medplum-typescript/src/model/fhir/NutritionOrder.ts
import { Annotation } from "./Annotation";
import { canonical } from "./canonical";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { NutritionOrder_EnteralFormula } from "./NutritionOrder_EnteralFormula";
import { NutritionOrder_OralDiet } from "./NutritionOrder_OralDiet";
import { NutritionOrder_Supplement } from "./NutritionOrder_Supplement";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface NutritionOrder {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    instantiatesCanonical: canonical[] | null | undefined;
    instantiatesUri: uri[] | null | undefined;
    instantiates: uri[] | null | undefined;
    status: code | null | undefined;
    intent: code | null | undefined;
    patient: Reference;
    encounter: Reference | null | undefined;
    dateTime: dateTime | null | undefined;
    orderer: Reference | null | undefined;
    allergyIntolerance: Reference[] | null | undefined;
    foodPreferenceModifier: CodeableConcept[] | null | undefined;
    excludeFoodModifier: CodeableConcept[] | null | undefined;
    oralDiet: NutritionOrder_OralDiet | null | undefined;
    supplement: NutritionOrder_Supplement[] | null | undefined;
    enteralFormula: NutritionOrder_EnteralFormula | null | undefined;
    note: Annotation[] | null | undefined;
}